packages/driver/src/cy/assertions.ts

```diff
@@ -430,99 +511,19 @@ export default {
-      // slice off everything after a ', but' or ' but ' for passing assertions, because
-      // otherwise it doesn't make sense:
-      // "expected <div> to have a an attribute 'href', but it was 'href'"
-      if (message && passed && butRe.test(message)) {
-        message = message.substring(0, message.search(butRe))
-      }
-
-      if (value && value.isSinonProxy) {
-        message = message.replace(stackTracesRe, '\n')
-      }
-
-      let obj = parseValueActualAndExpected(value, actual, expected)
-
-      if ($dom.isElement(value)) {
-        obj.$el = $dom.wrap(value)
-      }
-
-      // if we are simply verifying the upcoming
-      // assertions then do not immediately end or snapshot
-      // else do
-      if (verifying) {
-        obj._error = error
-      } else {
-        obj.end = true
-        obj.snapshot = true
-        obj.error = error
-      }
-
-      const isChildLike = (subject, current) => {
-        return (value === subject) ||
-          isDomSubjectAndMatchesValue(value, subject) ||
-          // if our current command is an assertion type
-          isAssertionType(current) ||
-          // are we currently verifying assertions?
-          (cy.state('upcomingAssertions') && cy.state('upcomingAssertions').length > 0) ||
-          // did the function have arguments
-          functionHadArguments(current)
-      }
-
-      _.extend(obj, {
-        name: 'assert',
-        // end:      true
-        // snapshot: true
-        message,
-        passed,
-        selector: value ? value.selector : undefined,
-        timeout: 0,
-        type (current, subject) {
-          // if our current command has arguments assume
-          // we are an assertion that's involving the current
-          // subject or our value is the current subject
-          return isChildLike(subject, current) ? 'child' : 'parent'
-        },
-
-        consoleProps: () => {
-          obj = { Command: 'assert' }
-
-          _.extend(obj, parseValueActualAndExpected(value, actual, expected))
-
-          return _.extend(obj,
-            { Message: message.replace(bTagOpen, '').replace(bTagClosed, '') })
-        },
-      })
-
-      // think about completely gutting the whole object toString
-      // which chai does by default, its so ugly and worthless
+    },
 
-      if (error) {
-        error.onFail = (err) => { }
-      }
-
-      Cypress.log(obj)
-
-      return null
-    }
-
-    const assert = function (...args) {
+    assert (...args) {
       // if we've temporarily overriden assertions
       // then just bail early with this function
       const fn = cy.state('overrideAssert') || assertFn
 
       return fn.apply(this, args)
-    }
-
-    return {
-      finishAssertions,
-
-      verifyUpcomingAssertions,
+    },
+  }
+}
 
-      assert,
-    }
-  },
+export interface IAssertions {
+  verifyUpcomingAssertions: ReturnType<typeof create>['verifyUpcomingAssertions']
```

Comment:
I intentionally didn't add it because it's not a public API. Check the code below:

https://github.com/cypress-io/cypress/blob/50d10700262645cc3953e647186e21eb117e4d52/packages/driver/src/cypress/cy.ts#L611-L612